packages/langium/test/parser/langium-parser.test.ts

```diff
@@ -89,6 +89,29 @@ describe('hidden node parsing', () => {
 
 });
 
+describe('parser error recovery', () => {
+
+    for (const [open, close] of [[1, 0], [2, 0], [3, 0], [2, 1], [3, 1], [3, 2]]) {
+        test(`recovers from lexer error with ${open} open and ${close} close parenthesis`, async () => {
+            const text = `
+                grammar Test
+                entry Model: value=Expr ';';
+                Expr: '(' Expr ')' | value=ID;
+                terminal ID: /[_a-zA-Z][\\w_]*/;
+                hidden terminal WS: /\\s+/;
+            `;
+            const services = await createServicesForGrammar({ grammar: text });
+            const opening = '('.repeat(open);
+            const closing = ')'.repeat(close);
+            const result = services.parser.LangiumParser.parse(`${opening}a${closing};`);
+            // Expect only one parser error independent of the number of missing closing parenthesis
+            console.log(result.parserErrors);
```

Comment:
I don't think you intended to leave this `console.log` here :)

Reply:
Thanks, yes I didn't :D